zerver/lib/webhooks/common.py

```diff
@@ -78,13 +78,16 @@ def check_send_webhook_message(
         # Some third-party websites (such as Atlassian's JIRA), tend to
         # double escape their URLs in a manner that escaped space characters
         # (%20) are never properly decoded. We work around that by making sure
-        # that the stream name is decoded on our end.
-        if unquote_stream:
+        # that the URL parameters are decoded on our end.
+        if unquote_url_parameters:
             stream = unquote(stream)
 
         if user_specified_topic is not None:
             topic = user_specified_topic
 
+        if unquote_url_parameters:
+            topic = unquote(topic)
```

Comment:
Should this be inside the  `user_defined_topic` block?

Reply:
Okay, yeah that makes more sense! :)